### PR TITLE
Update Raspberry Pi packages

### PR DESCRIPTION
**Pull request by leo60228**
###### Motivation for this change
The primary impetus was for Raspberry Pi Zero 2 W support, but there are minor changes affecting all hardware.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux (`pkgsCross.aarch64-multiplatform`)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by leo60228** [comment]
Ugh, wait, the `firmware` repo is tagged from the `stable` branch which doesn't have Zero 2 W support yet...

**Comment by leo60228** [comment]
I'll pick this back up once that's available.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -455,6 +455,8 @@ mapAliases ({
   links = links2; # added 2016-01-31
   linux_rpi0 = linuxKernel.kernels.linux_rpi1;
   linuxPackages_rpi0 = linuxKernel.packages.linux_rpi1;
+  linux_rpi02w = linuxKernel.kernels.linux_rpi3;
+  linuxPackages_rpi02w = linuxKernel.packages.linux_rpi3;
```
Comment (samueldr) [inline pkgs/top-level/aliases.nix:459]: I don't know if we should add this alias.

Comment (leo60228) [reply pkgs/top-level/aliases.nix:459]: This was for consistency with `linux_rpi0`, which was added in #68265.

Comment (samueldr) [reply pkgs/top-level/aliases.nix:459]: I still don't know, I don't care that much, it just might get confusing, the more device-specific kernel attrnames we accumulate.

Ideally I'd remove all vendor kernels from NIxpkgs!

**Comment by zhaofengli** [comment]
Boots on my Pi Zero 2 with the following [extra commits](https://github.com/zhaofengli/nixpkgs/commits/pi02):

- [raspberrypifw: 1.20211029 -> 1.20211118](https://github.com/zhaofengli/nixpkgs/commit/15d3873e8e409e7b919258059a7ddfa01c98e25e)
- [linux_rpi: 1.20211029 -> stable_20211118](https://github.com/zhaofengli/nixpkgs/commit/ec9d0e4c2e1377e0f79886916131b2d38453d93b)
- [ubootRaspberryPi3_{32,64}bit: Add patch to load correct DT for Pi Zero 2 (W)](https://github.com/zhaofengli/nixpkgs/commit/91ac58ccd95b732345c5d9b2b251e0cd8ac53960)
- [sd-image-aarch64.nix: Add config for Pi Zero 2 (W)](https://github.com/zhaofengli/nixpkgs/commit/fa9443d40364b6e9236d64f69f736e736db30ef3)
- The workaround in #126755

Could you cherry-pick?

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (zhaofengli) [reply pkgs/top-level/aliases.nix:459]: I think we should add it for consistency, and so that it's more straightforward for users.

**Comment by IvarWithoutBones** [comment]
Could you maybe make your own PR, @zhaofengli? It would be great to have support for this in master.

**Comment by zhaofengli** [comment]
> Could you maybe make your own PR, @zhaofengli? It would be great to have support for this in master.

Sure, I opened #156005.

**Comment by zhaofengli** [comment]
Closing since #156005 was merged with all changes here incorporated.

**Pull request closed without merging**